include/wil/cppwinrt_helpers.h

```diff
@@ -258,27 +274,27 @@ namespace wil
     */ 
     template<typename TSrc> auto to_vector(TSrc const& src)
     {
-        if constexpr (details::is_vector<TSrc>())
+        if constexpr (details::is_winrt_vector_like<TSrc>::value)
         {
             using T = decltype(src.GetAt(0));
             std::vector<T> result;
             if (auto expected = src.Size())
             {
                 result.resize(expected + 1, details::empty<T>());
                 auto actual = src.GetMany(0, result);
-                if (expected > actual)
+                if (actual > expected)

```

Comment:
Why not `actual != expected`? As-is, it allows fewer values to be returned

Reply:
If `actual < expected`, then it means that the vector shrunk while we were trying to capture it, but the capture was still successful: We got all the elements atomically.